### PR TITLE
Add fault injection macros for use in other packages

### PR DESCRIPTION
**Pull request by brawner**
This adds fault injection macros for use in `rcl_action` and `rcl_lifecycle`.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

**Comment by brawner** [comment]
Testing this PR branch against current ros 2 branches (`--packages-select rmw`)

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12117/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7115/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9866/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12023/)

File: rmw/src/topic_endpoint_info.c [hunk 1]
```diff
@@ -105,6 +108,8 @@ _rmw_topic_endpoint_info_copy_str(
   const char * str,
   rcutils_allocator_t * allocator)
 {
+  RCUTILS_CAN_RETURN_WITH_ERROR_OF(RMW_RET_INVALID_ARGUMENT);
```
Comment (Blast545) [inline rmw/src/topic_endpoint_info.c:111]: Should this be able of returning `BAD_ALLOC` as well? I don't know why the output of `rcutils_strdup` is not checked there

Comment (Blast545) [reply rmw/src/topic_endpoint_info.c:111]: Same with the remaining part of the API receiving an allocator

Comment (brawner) [reply rmw/src/topic_endpoint_info.c:111]: Opened #269 to add the new API and added a second `RCUTILS_CAN_RETURN_WITH_ERROR_OF` here.

**Comment by brawner** [comment]
Rebased onto master after merging #269, retesting

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12155/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7142/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9889/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12045/)

**Comment by brawner** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12157/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7145/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9891/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12047/)

**Comment by brawner** [comment]
A rebase did the trick

**Pull request merged**